### PR TITLE
[DEV-1770] Update online store table schema to use long table format

### PR DESCRIPTION
**Pull request by YS-L**
## Description

This updates the online store table schema to store precomputed values using a long table format. This allows setting a partition key (the aggregation result name column) in order to avoid errors caused by concurrent writes in delta tables.

## Type of Change

Label this pull request to place it under the correct category in Release Notes:

|        **Pull Request Label**         | **Category in Release Notes** |
|:-------------------------------------:|:-----------------------------:|
|       `enhancement`, `feature`        |          🚀 Features          |
| `bug`, `refactoring`, `bugfix`, `fix` |    🔧 Fixes & Refactoring     |
|       `build`, `ci`, `testing`        |    📦 Build System & CI/CD    |
|              `breaking`               |      💥 Breaking Changes      |
|            `documentation`            |       📝 Documentation        |
|            `dependencies`             |    ⬆️ Dependencies updates    |

## Checklist

- [ ] I have read the [`CODE_OF_CONDUCT.md`](https://github.com/featurebyte/featurebyte/blob/main/CODE_OF_CONDUCT.md) and [`CONTRIBUTING.md`](https://github.com/featurebyte/featurebyte/blob/main/CONTRIBUTING.md) guides.
- [ ] I have written tests for the changes made.
- [ ] I have written docstrings in [NumpyDoc format](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard)
- [ ] I have labeled my Pull Request correctly

### CODE REVIEW

File: featurebyte/query_graph/sql/adapter.py [hunk 1]
```diff
@@ -628,6 +707,99 @@ def create_table_as(cls, table_details: TableDetails, select_expr: Select) -> Ex
...
+        return value_column_expr
+
+    @classmethod
+    def online_store_pivot_finalise_value_column(
+        cls,
+        agg_result_name: str,
+        dtype: DBVarType,
+    ) -> Expression:
+        # Snowflake's PIVOT surrounds the pivoted fields with single quotes (')
+        agg_result_name_expr = expressions.Identifier(this=f"'{agg_result_name}'", quoted=True)
```
Comment (gee-senbong) [inline featurebyte/query_graph/sql/adapter.py:742]: Should we used `quoted_identifier` here?

File: featurebyte/query_graph/sql/specs.py [hunk 2]
```diff
@@ -217,11 +231,55 @@ def from_groupby_query_node(
...
+        pruned_graph, node_name_map, _ = prune_query_graph(
+            graph=graph, node=project_node, aggressive=True
+        )
+        pruned_node = pruned_graph.get_node_by_name(node_name_map[project_node.name])
+        op_struct = (
+            OperationStructureExtractor(graph=pruned_graph)
+            .extract(node=pruned_node)
+            .operation_structure_map[pruned_node.name]
+        )
+        assert len(op_struct.aggregations) == 1
```
Comment (gee-senbong) [inline featurebyte/query_graph/sql/specs.py:279]: Would be good to add some error message here for troubleshooting purpose.

File: featurebyte/sql/tile_schedule_online_store.py [hunk 3]
```diff
@@ -85,84 +86,85 @@ async def execute(self) -> None:
...
             )
 
             current_ts = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
 
             if not fs_table_exist_flag:
                 # feature store table does not exist, create table with the input feature sql
                 create_sql = construct_create_table_query(
-                    fs_table, f"select {entities_fname_str} from ({f_sql})", session=self._session
+                    fs_table,
+                    f"select {column_names} from ({f_sql})",
```
Comment (gee-senbong) [inline featurebyte/sql/tile_schedule_online_store.py:112]: Should we use sqlglot here?

**Review by gee-senbong — APPROVED** [review]
LGTM

**Review by kchua78 — APPROVED** [review]

File: featurebyte/sql/tile_schedule_online_store.py (hunk 3 above)
Comment (YS-L) [reply featurebyte/sql/tile_schedule_online_store.py:112]: Ideally this should be using sqlglot. To minimise unrelated changes I think it's better to refactor that in separate PRs.

**Pull request merged**